sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientIntegrationTest.java

```diff
@@ -69,71 +69,123 @@ public void listPurchasedPhoneNumbers(HttpClient httpClient) {
+                searchOptions,
+                Context.NONE));
+        }
         return setPollInterval(getClientWithConnectionString(httpClient, testName).beginSearchAvailablePhoneNumbers(
             COUNTRY_CODE,
             PhoneNumberType.TOLL_FREE,
             PhoneNumberAssignmentType.APPLICATION,
-            capabilities,
-            searchOptions,
-            Context.NONE));
+            capabilities));
     }
 
-    private SyncPoller<PhoneNumberOperation, Void> beginPurchasePhoneNumbersHelper(HttpClient httpClient, String searchId, String testName) {
-        return setPollInterval(this.getClientWithConnectionString(httpClient, testName)
+    private SyncPoller<PhoneNumberOperation, Void> beginPurchasePhoneNumbersHelper(HttpClient httpClient, String searchId, String testName, boolean withContext) {
+        if (withContext) {
+            return setPollInterval(this.getClientWithConnectionString(httpClient, testName)
             .beginPurchasePhoneNumbers(searchId, Context.NONE));
+        }
+        return setPollInterval(this.getClientWithConnectionString(httpClient, testName)
+        .beginPurchasePhoneNumbers(searchId));
     }
 
-    private SyncPoller<PhoneNumberOperation, Void> beginReleasePhoneNumberHelper(HttpClient httpClient, String phoneNumber, String testName) {
+    private SyncPoller<PhoneNumberOperation, Void> beginReleasePhoneNumberHelper(HttpClient httpClient, String phoneNumber, String testName, boolean withContext) {
         if (getTestMode() == TestMode.PLAYBACK) {
             phoneNumber = "+REDACTED";
         }
-        return setPollInterval(this.getClientWithConnectionString(httpClient, testName)
+        if (withContext) {
+            return setPollInterval(this.getClientWithConnectionString(httpClient, testName)
             .beginReleasePhoneNumber(phoneNumber, Context.NONE));
+        }
+        return setPollInterval(this.getClientWithConnectionString(httpClient, testName)
+        .beginReleasePhoneNumber(phoneNumber));
     }
 
-    private SyncPoller<PhoneNumberOperation, PurchasedPhoneNumber> beginUpdatePhoneNumberCapabilitiesHelper(HttpClient httpClient, String phoneNumber, String testName) {
+    private SyncPoller<PhoneNumberOperation, PurchasedPhoneNumber>
+        beginUpdatePhoneNumberCapabilitiesHelper(HttpClient httpClient, String phoneNumber, String testName, boolean withContext) {
         PhoneNumberCapabilitiesRequest capabilitiesUpdateRequest = new PhoneNumberCapabilitiesRequest();
         capabilitiesUpdateRequest.setCalling(PhoneNumberCapabilityType.INBOUND);
         capabilitiesUpdateRequest.setSms(PhoneNumberCapabilityType.INBOUND_OUTBOUND);
-        return setPollInterval(this.getClientWithConnectionString(httpClient, testName)
+        if (withContext) {
+            return setPollInterval(this.getClientWithConnectionString(httpClient, testName)
             .beginUpdatePhoneNumberCapabilities(phoneNumber, capabilitiesUpdateRequest, Context.NONE));
```

Comment:
Question: if `withContext` is true, why pass in `Context.NONE`? It seems opposite to me.

Reply:
Context.NONE is still a type of context. `withContext` means that we are passing in a context type to the function rather than not passing a context at all.